arbitrum/apibackend.go

```diff
@@ -84,8 +85,95 @@ func (a *APIBackend) SuggestGasTipCap(ctx context.Context) (*big.Int, error) {
 	return big.NewInt(0), nil // there's no tips in L2
 }
 
-func (a *APIBackend) FeeHistory(ctx context.Context, blockCount int, lastBlock rpc.BlockNumber, rewardPercentiles []float64) (*big.Int, [][]*big.Int, []*big.Int, []float64, error) {
-	return nil, nil, nil, nil, errors.New("not implemented")
+func (a *APIBackend) FeeHistory(
+	ctx context.Context,
+	blocks int,
+	newestBlock rpc.BlockNumber,
+	rewardPercentiles []float64,
+) (*big.Int, [][]*big.Int, []*big.Int, []float64, error) {
+
+	if core.GetArbOSComputeRate == nil {
+		return nil, nil, nil, nil, errors.New("ArbOS not installed")
+	}
+
+	latestBlock := rpc.BlockNumber(a.CurrentBlock().NumberU64())
+	if newestBlock == rpc.LatestBlockNumber || newestBlock == rpc.PendingBlockNumber {
+		newestBlock = latestBlock
+	}
+	if newestBlock > latestBlock {
+		newestBlock = latestBlock
+	}
+
+	maxFeeHistory := int(a.b.config.FeeHistoryMaxBlockCount)
+	if blocks > maxFeeHistory {
+		log.Warn("Sanitizing fee history length", "requested", blocks, "truncated", maxFeeHistory)
+		blocks = maxFeeHistory
+	}
+	if blocks < 1 {
+		// returning with no data and no error means there are no retrievable blocks
+		return common.Big0, nil, nil, nil, nil
+	}
+
+	// don't attempt to include blocks before genesis
+	if rpc.BlockNumber(blocks) > newestBlock {
+		blocks = int(newestBlock + 1)
+	}
+	oldestBlock := int(newestBlock) + 1 - blocks
```

Comment:
For this I think we'll need to be careful to have the max depth be the nitro genesis block if we're starting from a state port